node/router/src/lib.rs

```diff
@@ -368,46 +343,31 @@ impl<N: Network> Router<N> {
         self.peer_pool.read().iter().filter(|(_, peer)| peer.is_connected()).count()
     }
 
-    /// Returns the number of connected validators.
-    pub fn number_of_connected_validators(&self) -> usize {
-        self.peer_pool.read().values().filter(|peer| peer.node_type() == Some(NodeType::Validator)).count()
-    }
-
-    /// Returns the number of connected provers.
-    pub fn number_of_connected_provers(&self) -> usize {
-        self.peer_pool.read().values().filter(|peer| peer.node_type() == Some(NodeType::Prover)).count()
-    }
-
-    /// Returns the number of connected clients.
-    pub fn number_of_connected_clients(&self) -> usize {
-        self.peer_pool.read().values().filter(|peer| peer.node_type() == Some(NodeType::Client)).count()
-    }
-
     /// Returns the number of candidate peers.
     pub fn number_of_candidate_peers(&self) -> usize {
         self.peer_pool.read().values().filter(|peer| matches!(peer, Peer::Candidate(_))).count()
     }
 
-    /// Returns the number of restricted peers.
-    pub fn number_of_restricted_peers(&self) -> usize {
-        self.peer_pool
-            .read()
-            .values()
-            .filter(|peer| matches!(peer, Peer::Candidate(peer) if peer.restricted.is_some()))
-            .count()
-    }
-
     /// Returns the connected peer given the peer IP, if it exists.
     pub fn get_connected_peer(&self, ip: &SocketAddr) -> Option<ConnectedPeer<N>> {
         if let Some(Peer::Connected(peer)) = self.peer_pool.read().get(ip) { Some(peer.clone()) } else { None }
     }
 
     /// Returns the connected peers.
     pub fn get_connected_peers(&self) -> Vec<ConnectedPeer<N>> {
+        self.filter_connected_peers(|_| true)
+    }
+
+    pub fn filter_connected_peers<P: FnMut(&ConnectedPeer<N>) -> bool>(
+        &self,
+        mut predicate: P,
+    ) -> Vec<ConnectedPeer<N>> {
         self.peer_pool
             .read()
             .values()
-            .filter_map(|peer| if let Peer::Connected(p) = peer { Some(p.clone()) } else { None })
+            .filter_map(|peer| if let Peer::Connected(peer) = peer { Some(peer) } else { None })
+            .filter(|peer| predicate(peer))
```

Comment:
Would it make sense to merge these filters? Let chains are stable now. 
```suggestion
            .filter_map(|peer| if let Peer::Connected(peer) = peer && predicate(peer) { Some(peer) } else { None })
```

Reply:
oh yeah, good point